Create articles, people and events Layout Builder blocks.

joewhitsitt opened the pull request:
Resolves #1789, resolves #2071 

Introduces three new custom block types (articles, people, events) that are basically ports of the paragraph versions.

Created https://github.com/uiowa/uiowa/issues/2058 to address alternative layout/style options

## To Do

- [x] Fix caching issues with collection heading
- [x] Remove border-top, consistent top margin
- [x] Recreate and share fields between articles/people and events block
- [x] Add collection heading to events block
- [x] Add LB styles to events block
- [x] Add event location to teaser and style appropriately for hide-details LB style.
- [ ] ~~Update event styles to match existing in UIDS.~~
- [x] Update sitenow_events instance date formats.

## To Test

- Using a v3 site with articles, people and events (more than one of each), pull/sync. 
  - The default site can reinstalled locally as v3 by running `blt drupal:install`.
- Compile frontend SASS via ` blt frontend` or `yarn workspace uids_base gulp`.
- As a webmaster ensure there are three new blocks available (Articles, People, Events). Configure/place as you like.
- Within the layout interface and the rendered content (depending on content available) you should see a list of content and your configuration should match the output (hiding images, displaying a certain number, displaying a pager if more than number of items, filtered by tag, more link with custom path, etc.).
- Set heading field and make sure the teaser markup includes the correct heading tag (not the h5 class).
- Out of scope: there is intentionally no grid, masonry or other layout options included with this PR.
- For events, test out a single event page by changing the link path config at Admin -> Events. (drush cr afterwards).

I have not built this with collegiate split in mind as I am sure there are at least permission updates involved.

joewhitsitt committed:
more fields, basic styles, more repurposed spaghetti

joewhitsitt committed:
Merge branch 'main' into i1789-views-block

joewhitsitt committed:
initial people block work

joewhitsitt committed:
styles, layout builder style option

joewhitsitt committed:
hide on lists option was never accounted for

joewhitsitt committed:
get hide descriptions, details setup

cory-skeers committed:
Merge branch 'main' into i1789-views-block

cory-skeers committed:
Update field content to default to minimal html formatting.

joewhitsitt committed:
Merge branch 'main' into i1789-views-block

joewhitsitt committed:
add #states to more link, path

cory-skeers reviewed (approved):
Reviewed and tested out on uiowa.edu using the default Keep Moving Forward! and Herky default content, and some new test articles/people.

Things look to be working well, and matches v2 implementation. Can get some strange hierarchies in sections with its `<h3>` tags, but that's a bigger question, and not a new one.

cory-skeers committed:
Merge branch 'main' of github.com:uiowa/uiowa into main

cory-skeers committed:
Merge main.

cory-skeers committed:
Removed redundant block process addition from main merge.

joewhitsitt committed:
Merge branch 'main' into i1789-views-block

joewhitsitt committed:
Merge branch 'main' into i1789-views-block

joewhitsitt committed:
add configurable heading field similar to collections, slider

joewhitsitt committed:
hide source link with details option

joewhitsitt committed:
help text

joewhitsitt committed:
fix people by tag logic

joewhitsitt committed:
merge main

joewhitsitt committed:
coding standard

Commit:
Merge branch 'main' into i1789-views-block

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Export initial uiowa_events block config

Commit:
Change paragraphs events block to match prefixed allowed values callback

Commit:
Refactor sitenow_events dynamic field allowed values callback

Update docblock, fix optional arg before required, prefix function
with sitenow_events, move helper function inline with callback since
it is not used elsewhere, update switch cases to work with uiowa_events
block field.

Commit:
Fix hook docblock in LBC

Commit:
Process events more link states on add/update LB blocks

Commit:
Use single quotes

Commit:
Move "private" functions below "public" in sitenow_events.module

Commit:
PHPCS

richardbporter committed:
Merge branch 'main' into uiowa-events-block

Commit:
Preprocess uiowa_events block to get data rendering

pyrello committed:
Merge branch 'main' into i1789-views-block

Commit:
Create basic events library for list style

richardbporter committed:
Merge branch 'main' into uiowa-events-block

Commit:
Fix more link

Commit:
Update more link help text

Commit:
Loosen comparison of more link value

Commit:
Fix events more link empty value

Commit:
Fix last event border styles

richardbporter committed:
Merge branch 'main' into uiowa-events-block

Commit:
Remove unnecessary event config items and nesting

Commit:
Merge branch 'uiowa-events-block' of github.com:uiowa/uiowa into uiowa-events-block

Commit:
Remove trailing slash of base endpoint

Commit:
Reduce cache to 5 minutes

Commit:
Change base endpoint to include trailing slash - breaks everything

Commit:
Simplify filter api requests

Commit:
Update event/events block descriptions

Commit:
Add dependency on sitenow_events

Commit:
Safely replace string comparions ternary with expression result

Commit:
Remove duplicate semicolon

Commit:
Fix typo

Commit:
Remove unused local var

Commit:
Clean up events load and docs

Commit:
Document todos

Commit:
Add todo

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Fix single event all day time

Commit:
Remove instance values helper function and pass params directly into load

Commit:
Use UrlHelper::BuildQuery instead of our own and reformat multivalue filters

joewhitsitt committed:
refactor to share fields and fix pager/items issue with people

Commit:
Refactor sitenow_events_load to work with other endpoints and let theme functions handle formatting data

richardbporter committed:
Merge branch 'main' into uiowa-events-block

Commit:
Return empty data if request fails

Commit:
Check if request isset

Commit:
Add slash back to default config base endpoint

Commit:
Refactor select list group allowed values callback

Change to use sitenow_events_load instead of one-off API calls. This
caches the data for better performance and only makes the API call once
per field as opposed to every time the field callback fires. Also
refactor the option select list helper functions to simplify and use
optgroups instead of text.

Commit:
PHPCS

Commit:
Remove unused var

Commit:
Set _title_callback in dynamic route callback

Commit:
Fix single event page title which was not working, ever

joewhitsitt committed:
Merge branch 'main' into i1789-views-block

Commit:
Reimplement recursive option lists, whoops

richardbporter committed:
Merge branch 'main' into uiowa-events-block

Commit:
Export places block field

Commit:
Wire up places filter in events block

Commit:
Rename places field to place since it accepts only one value

Commit:
Update switch after place field rename

Commit:
Note todo about replacing complicated recursive taxonomy parsing

Commit:
PHPCS

Commit:
Merge branch 'uiowa-events-block' into i1789-views-block

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Add collection heading size to events block

Commit:
Preprocess heading size var for each event and pass to template

richardbporter committed:
Merge branch 'main' into i1789-views-block

joewhitsitt committed:
make each block similar on top

joewhitsitt committed:
Merge branch 'i1789-views-block' of github.com:uiowa/uiowa into i1789-views-block

joewhitsitt committed:
remove cache from view blocks

Commit:
Allow hide-details/description/image in events block

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Add block cache tags to view result entities for desired invalidation

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Remove entity->save

Commit:
Setup custom cache tag assoc for people, move between view->excecute and render

joewhitsitt committed:
rename and share fields between blocks

joewhitsitt committed:
Merge branch 'i1789-views-block' of github.com:uiowa/uiowa into i1789-views-block

Commit:
Revert "Fix single event all day time"

This reverts commit 704df4a10aaa08dcd19611450cca6e6a3e690961.

Commit:
Fix day comparison in combo with all day events

richardbporter committed:
Merge branch 'main' into i1789-views-block

Commit:
Try to account for all possible date instances

Commit:
Change single event data structure to event

Commit:
Append canceled badge to event teaser

Commit:
Update comment

Commit:
PHPCS

Commit:
Move events lib, refactor event teaser canceled

Commit:
Style teaser location and hide with hide-details

richardbporter committed:
Merge branch 'main' into i1789-views-block

richardbporter commented on config/default/block_content.type.uiowa_articles.yml:
```diff
@@ -0,0 +1,8 @@
+uuid: cbea77a6-0e50-488a-8d6c-c228b11dc1b9
+langcode: en
+status: true
+dependencies: {  }
```
I guessing this has a hard dependency on sitenow_articles, as does sitenow_people.

pyrello replied:
Added.

richardbporter commented on docroot/modules/custom/layout_builder_custom/layout_builder_custom.module:
```diff
@@ -172,17 +172,33 @@ function _layout_builder_custom_process_element(array $element, FormStateInterfa
+      break;
+
+    case 'inline_block:uiowa_articles':
+    case 'inline_block:uiowa_people':
```
We should make this work for events too.

Commit:
Fix single event image

Commit:
Move events more link states alter to LBC

Commit:
Move more link right below boolean on events block

Commit:
Support virtual event location

richardbporter commented on docroot/modules/custom/sitenow_events/css/single-event.css:
```diff
@@ -6,6 +6,10 @@
   }
 }
 
+.sitenow-event .event-field {
```
Just noticed that `.sitenow-event` seems like the wrong selector for an event page because it doesn't exist. However, changing it to `.sitenow-event-single`, which exists, really borks the display.

pyrello replied:
It seems like the event page looks pretty good now. We may want to have @bspeare take a look and refine it a bit when he gets back, but this looks like a great starting point!

Commit:
Lazy load images

joewhitsitt committed:
fix merge

GaryRidgway committed:
minor style changes

GaryRidgway committed:
small margin change

joewhitsitt committed:
tidy up single event

richardbporter committed:
Merge branch 'main' into i1789-views-block

pyrello committed:
Merge branch 'main' into i1789-views-block

pyrello committed:
Added dependencies to config for articles and people blocks.

pyrello reviewed (approved):
This looks good to me. There may be a little clean-up to do, but I think the data structure is sound.
